plugins/experimental/filter_body/filter_body.cc

```diff
@@ -0,0 +1,771 @@
+          // will cause ATS to generate the error response
+          TSVIONBytesSet(data->output_vio, 0);
+          TSVIOReenable(data->output_vio);
+
+          // Consume all remaining input
+          int64_t remaining = TSIOBufferReaderAvail(reader);
+          if (remaining > 0) {
+            TSIOBufferReaderConsume(reader, remaining);
+          }
+          TSVIONDoneSet(write_vio, TSVIONBytesGet(write_vio));
+
+          // Signal write complete
+          TSContCall(TSVIOContGet(write_vio), TS_EVENT_VCONN_WRITE_COMPLETE, write_vio);
+          return 0;
+        }
+
+        // Zero-copy: copy data through to output
+        TSIOBufferCopy(data->output_buffer, reader, avail, 0);
+        TSIOBufferReaderConsume(reader, avail);
+        TSVIONDoneSet(write_vio, TSVIONDoneGet(write_vio) + avail);
+      }
+    }
+
+    // Check if we're done
+    if (TSVIONTodoGet(write_vio) > 0) {
+      if (towrite > 0) {
+        TSVIOReenable(data->output_vio);
+        TSContCall(TSVIOContGet(write_vio), TS_EVENT_VCONN_WRITE_READY, write_vio);
+      }
+    } else {
+      TSVIONBytesSet(data->output_vio, TSVIONDoneGet(write_vio));
+      TSVIOReenable(data->output_vio);
+      TSContCall(TSVIOContGet(write_vio), TS_EVENT_VCONN_WRITE_COMPLETE, write_vio);
+    }
+    break;
+  }
+  }
+
+  return 0;
+}
+
+// Create transform continuation
+TSVConn
+create_transform(TSHttpTxn txnp, const FilterConfig *config, const std::vector<const Rule *> &active_rules)
+{
+  TSVConn connp = TSTransformCreate(transform_handler, txnp);
+
+  auto *data         = new TransformData();
+  data->txnp         = txnp;
+  data->config       = config;
+  data->active_rules = active_rules;
+
+  // Pre-allocate lookback buffer
+  if (config->max_lookback > 0) {
+    data->lookback.reserve(config->max_lookback);
+  }
+
+  TSContDataSet(connp, data);
+  return connp;
+}
+
+// Hook handler for response rules (request rules are handled directly in TSRemapDoRemap)
+int
+hook_handler(TSCont contp, TSEvent event, void *edata)
+{
+  TSHttpTxn           txnp   = static_cast<TSHttpTxn>(edata);
+  const FilterConfig *config = static_cast<const FilterConfig *>(TSContDataGet(contp));
+
+  if (config == nullptr) {
+    TSHttpTxnReenable(txnp, TS_EVENT_HTTP_CONTINUE);
+    return 0;
+  }
+
+  TSMBuffer bufp;
+  TSMLoc    hdr_loc;
+
+  std::vector<const Rule *> active_rules;
+
+  if (event == TS_EVENT_HTTP_READ_RESPONSE_HDR) {
+    // Check response rules
+    if (TSHttpTxnServerRespGet(txnp, &bufp, &hdr_loc) != TS_SUCCESS) {
+      TSHttpTxnReenable(txnp, TS_EVENT_HTTP_CONTINUE);
+      return 0;
+    }
+
+    // Need client request for method check
+    TSMBuffer req_bufp;
+    TSMLoc    req_hdr_loc;
+    if (TSHttpTxnClientReqGet(txnp, &req_bufp, &req_hdr_loc) != TS_SUCCESS) {
+      TSHandleMLocRelease(bufp, TS_NULL_MLOC, hdr_loc);
+      TSHttpTxnReenable(txnp, TS_EVENT_HTTP_CONTINUE);
+      return 0;
+    }
+
+    for (const auto &rule : config->response_rules) {
```

Comment:
Can you change all the const identifiers to right side const? Thus `auto const &rule` instead of `cosnt auto &rule`.